pkg/appender/ingest.go

```diff
@@ -373,3 +383,22 @@ func setError(mc *MetricsCache, metric *MetricState, err error) {
 	metric.setError(err)
 	mc.lastError = err
 }
+
+const (
+	errorCodeString              = "ErrorCode"
+	falseConditionOuterErrorCode = "16777244"
+	falseConditionInnerErrorCode = "16777245"
+)
+
+// Check if the current error was caused specifically because the condition was evaluated to false.
+func isFalseConditionError(err error) bool {
```

Comment:
as i said yesterday, i also have this method for my delete PR. so let's put this peace of code in some common utility pkg

Reply:
np, moving to `utils/misc.go`